docs/architecture/maui/unit-testing.md

```diff
@@ -1,9 +1,223 @@
+    await orderViewModel.InitializeAsync(order);
+
+    Assert.NotNull(orderViewModel.Order);
+}
+```
+
+This unit test checks that the `Order` property of the `OrderDetailViewModel` instance will have a value after the `InitializeAsync` method has been invoked. The InitializeAsync`` method is invoked when the view model's corresponding view is navigated to. For more information about navigation, see [Navigation](navigation.md).
+
+When the `OrderDetailViewModel` instance is created, it expects an `IOrderService` instance to be specified as an argument. However, the `OrderService` retrieves data from a web service. Therefore, an `OrderMockService` instance, a mock version of the `OrderService` class, is specified as the argument to the `OrderDetailViewModel` constructor. Then, mock data is retrieved rather than communicating with a web service when the view model's `InitializeAsync` method is invoked, which uses `IOrderService` operations.
+
+## Testing INotifyPropertyChanged implementations
+
+Implementing the `INotifyPropertyChanged` interface allows views to react to changes that originate from view models and models. These changes are not limited to data shown in controls -- they are also used to control the view, such as view model states that cause animations to be started or controls to be disabled.
+
+Properties that can be updated directly by the unit test can be tested by attaching an event handler to the `PropertyChanged` event and checking whether the event is raised after setting a new value for the property. The following code example shows such a test:
+
+```csharp
+[Fact]
+public async Task SettingOrderPropertyShouldRaisePropertyChanged()
+{
+    bool invoked = false;
+    var orderService = new OrderMockService();
+    var orderViewModel = new OrderDetailViewModel(orderService);
+
+    orderViewModel.PropertyChanged += (sender, e) =>
+    {
+        if (e.PropertyName.Equals("Order"))
+            invoked = true;
+    };
+    var order = await orderService.GetOrderAsync(1, GlobalSetting.Instance.AuthToken);
+    await orderViewModel.InitializeAsync(order);
+
+    Assert.True(invoked);
+}
+```
+
+This unit test invokes the `InitializeAsync` method of the `OrderViewModel` class, which causes its `Order` property to be updated. The unit test will pass, provided that the `PropertyChanged` event is raised for the `Order` property.
+
+## Testing message-based communication
+
+View models that use the `MessagingCenter` class to communicate between loosely-coupled classes can be unit tested by subscribing to the message being sent by the code under test, as demonstrated in the following code example:
+
+```csharp
+[Fact]
+public void AddCatalogItemCommandSendsAddProductMessageTest()
+{
+    bool messageReceived = false;
+    var catalogService = new CatalogMockService();
+    var catalogViewModel = new CatalogViewModel(catalogService);
+
+    .NET MAUI.MessagingCenter.Subscribe<CatalogViewModel, CatalogItem>(
```

Comment:
```suggestion
    MessagingCenter.Subscribe<CatalogViewModel, CatalogItem>(
```

Not sure what happened here, please verify that my suggestion is correct, but this will not compile.

Reply:
This was an accidental replacement. Updated to just `MessagignCenter...`